File: bite_323.py

# Online Python - IDE, Editor, Compiler, Interpreter
from typing import Any
from typing import Iterable
from typing import Set
from functools import reduce

def intersection(*args: Iterable) -> Set[Any]:
    iterables = [set(iterable) for iterable in args if iterable]
    if iterables:
        return set.intersection(*iterables)

File: bite_323.js
function intersection(...args) {
  const iterables = args.filter(Boolean).map(iterable => new Set(iterable));
  if (iterables.length > 0) {
    return new Set([...iterables.reduce((a, b) => new Set([...a].filter(x => b.has(x))))]);
  } else {
    return new Set();
  }
}
